src/scripts/main.js

```diff
@@ -1,3 +1,324 @@
 'use strict';
 
-// write code here
+// TABLE_VARIABLES
+const table = document.querySelector('table');
+const tHead = table.tHead;
+const tBody = table.tBodies[0];
+
+// #region SORTING
+// SORTING FUNCTIONS
+const valueConversionFunctions = {
+  age: parseInt,
+  salary: window.convertSalartyToNumber,
+};
+
+const findSuitableSortingFunction = (
+  index,
+  order,
+  transformFunc = (x) => x,
+) => {
+  return (a, b) => {
+    const valueA = transformFunc(window.getValueByCellIndex(a, index));
+    const valueB = transformFunc(window.getValueByCellIndex(b, index));
+
+    if (typeof valueA === 'string' && typeof valueB === 'string') {
+      return order * valueA.localeCompare(valueB);
+    }
+
+    return order * (valueA - valueB);
+  };
+};
+
+const getSortingFunction = (index, colHeader, ascOrder) => {
+  const transformFunc = valueConversionFunctions[colHeader];
+
+  return ascOrder
+    ? findSuitableSortingFunction(index, 1, transformFunc)
+    : findSuitableSortingFunction(index, -1, transformFunc);
+};
+
+// TOGGLE ASC/DESC ORDER LOGIC
+const ascDescOrderUtil = {
+  element: undefined,
+  ascOrder: undefined, // true -> asc, false -> desc
+
+  clickIsMade: (target) => {
+    if (target === ascDescOrderUtil.element) {
```

Comment:
In the `clickIsMade` function, you are directly mutating the `ascDescOrderUtil` object. It would be better to avoid direct mutation of objects. Consider using a function to update the state of the object.